Felienne/hedy
# Add Keywords to the translate page

**Description** 
With this pull request, it will be possible to give a translation for all the keyword tokens on the translation page. Then, these translations can be downloaded in a yaml file and can be added to coursedata/keywords/. I have added a function in the file yaml_transformer.py, which can create lark files in the grammars directory for yaml files in the coursedata/keywords/ directory. There is an option to overwrite all these lark files or just add new lark files without overwriting the other ones. The approach in the function I have added in yaml_transformer.py, differs a bit from the existing transforming functions. This is because I could not get these function to work on my device (I kept getting import errors while all packages in requirements are installed). I don't know if this happened because of my device or because of some updates in modules. Anyway, I hope that the way the 'transform_yaml_to_lark' function works is also fine. 

I have updated TRANSLATING.md accordingly. 

Fixes issue #1109 

**How to test**

- Go to http://localhost:5000/translate/en/new
- Fill in the translations for the keywords
- Download keywords/new.yaml 
- Add it to the directory coursedata/keywords
- Run yaml_transformer.py (at least the function transform_yaml_to_lark)
- If all goes well, a keywords-new.lark file should be added to the directory grammars
